compose/config/config.py

```diff
@@ -232,20 +309,20 @@ def merge_services(base, override):
             for name in all_service_names
         }
 
-    config_file = process_config_file(config_details.config_files[0])
-    for next_file in config_details.config_files[1:]:
-        next_file = process_config_file(next_file)
-
+    config_file = config_files[0]
+    for next_file in config_files[1:]:
         config = merge_services(config_file.config, next_file.config)
         config_file = config_file._replace(config=config)
 
     return build_services(config_file)
 
 
-def process_config_file(config_file, service_name=None):
-    validate_top_level_object(config_file)
-    processed_config = interpolate_environment_variables(config_file.config)
-    validate_against_fields_schema(processed_config, config_file.filename)
+def process_config_file(config_file, version, service_name=None):
+    validate_top_level_service_objects(config_file, version)
+    processed_config = interpolate_environment_variables(config_file.config, version)
```

Comment:
I notice both of these functions have a very similar line added to them to lookup the service_dicts. 

It would be nice to abstract that away a little better, and remove the duplication.  Maybe either a function `get_service_dict(config_file, version)` called from here, and only pass in the service_dicts to each of these?

Or maybe a property or method on the `ConfigFile` class for returning them?